### PR TITLE
SMC message handling for messages from XVISOR

### PR DESCRIPTION
**Pull request by mkanalatc**
Initial commit to process SMC messages from Xvisor.

### CODE REVIEW

File: core/pta/tcta.c [hunk 1]
```diff
@@ -1,5 +1,9 @@
 
 #include <kernel/pseudo_ta.h>
+#include <mm/core_memprot.h>
```
Comment (rdevarajantct) [inline core/pta/tcta.c]: Don't think you need all these. Just mm/core_mmu.h is enough

File: core/pta/tcta.c [hunk 2]
```diff
@@ -17,8 +53,32 @@ uint32_t trustcore_smc_handler(uint8_t sub_id, uint64_t arg1,
...
+    vaddr_t *va;
+    tc_shmem_t *tct_shm;
 
+    switch(sub_id)
+    {
+	    case TC_SHMEM_TEE_SUB_ID:
+		    va = (vaddr_t)core_mmu_add_mapping(MEM_AREA_RAM_NSEC,arg1,sizeof(tc_shmem_t));
+		    EMSG("VA 0x%" PRIx64 " ", va);
+		    if (!va)
+                        goto bad_addr;
```
Comment (rdevarajantct) [inline core/pta/tcta.c]: goto is not really required here. Just return if the va is not NULL

File: core/pta/tcta.c [hunk 3]
```diff
@@ -17,8 +53,32 @@ uint32_t trustcore_smc_handler(uint8_t sub_id, uint64_t arg1,
...
+    
+    /* Virtual address of the shared memory */
+    vaddr_t *va;
+    tc_shmem_t *tct_shm;
 
+    switch(sub_id)
+    {
+	    case TC_SHMEM_TEE_SUB_ID:
+		    va = (vaddr_t)core_mmu_add_mapping(MEM_AREA_RAM_NSEC,arg1,sizeof(tc_shmem_t));
+		    EMSG("VA 0x%" PRIx64 " ", va);
```
Comment (rdevarajantct) [inline core/pta/tcta.c]: I think this message is not required. Just move the Shared memory address printing message here from line 69

File: core/pta/tcta.c [hunk 4]
```diff
@@ -17,8 +53,32 @@ uint32_t trustcore_smc_handler(uint8_t sub_id, uint64_t arg1,
...
+		    if (!va)
+                        goto bad_addr;
+                    tct_shm = (tc_shmem_t *)va;
+		    EMSG("TCT Shared memory at 0x%" PRIx64 " ", tct_shm);
+		    EMSG("tee_data_ready arrived as:%d", tct_shm->tee_data_ready);
```
Comment (rdevarajantct) [inline core/pta/tcta.c]: this is also not required.

File: core/pta/tcta.c [hunk 5]
```diff
@@ -17,8 +53,32 @@ uint32_t trustcore_smc_handler(uint8_t sub_id, uint64_t arg1,
...
+		    break;
+	    case TC_NOTIFY_DATA_TEE_SUB_ID:
+		    break;
+	    case TC_POLL_DATA_TEE_SUB_ID:
+		    break;
+    }
     return 0;
+    bad_addr:
+        EMSG("Bad arg address 0x%"PRIxPA, arg1);
```
Comment (rdevarajantct) [inline core/pta/tcta.c]: as mentioned in previous comment move this to a conditional above

**Review by rdevarajantct — CHANGES_REQUESTED** [review]

File: core/pta/tcta.c (hunk 1 above)
Comment (mkanalatc) [reply core/pta/tcta.c]: Done. Changed  to <mm/core_mmu.h>

File: core/pta/tcta.c (hunk 2 above)
Comment (mkanalatc) [reply core/pta/tcta.c]: Done.  Updated it.

File: core/pta/tcta.c (hunk 3 above)
Comment (mkanalatc) [reply core/pta/tcta.c]: Removed this print.

File: core/pta/tcta.c (hunk 4 above)
Comment (mkanalatc) [reply core/pta/tcta.c]: Removed it.

**Review by rdevarajantct — APPROVED** [review]

**Pull request merged**